lib/bus.rb

```diff
@@ -24,12 +33,28 @@ def self.[](key)
     redis.get(key)
   end
 
+  def self.store_keys(keys)
+    redis.pipelined do |pipeline|
+      keys.each do |key, value|
+        pipeline.set(key, value)
+        pipeline.expire(key, ttl_for(key))
+      end
+    end
+
+    true
+  end
+
   def self.[]=(key, value)
-    redis.pipelined do
-      redis.set(key, value)
-      redis.expire(key, TTL)
+    redis.pipelined do |pipeline|
+      pipeline.set(key, value)
+      pipeline.expire(key, ttl_for(key))
     end
 
     value
   end
+
+  def self.ttl_for(key)
+    prefix = key.split(':').first
```

Comment:
split(':')[0]

Reply:
Is [0] more efficient?

Reply:
no, but i prefer it because it's more concise